ui/app/controllers/jobs/job/allocations.js

```diff
@@ -53,4 +104,58 @@ export default class AllocationsController extends Controller.extend(
   gotoAllocation(allocation) {
     this.transitionToRoute('allocations.allocation', allocation);
   }
+
+  get optionsAllocationStatus() {
+    return [
+      { key: 'queued', label: 'Queued' },
+      { key: 'starting', label: 'Starting' },
+      { key: 'running', label: 'Running' },
+      { key: 'complete', label: 'Complete' },
+      { key: 'failed', label: 'Failed' },
+      { key: 'lost', label: 'Lost' },
+    ];
+  }
+
+  @computed('model.allocations.[]', 'selectionClient')
+  get optionsClients() {
+    const clients = Array.from(new Set(this.model.allocations.mapBy('node.shortId'))).compact();
```

Comment:
We are repeating this multiple times in this file. I understand that this pattern is used in other places as well `controllers/clients/index.js` for example. It looks like we can create a computed-macro out of this pattern.